src/Microsoft.Restier.Core/Query/DefaultQueryHandler.cs

```diff
@@ -146,17 +151,19 @@ public override Expression Visit(Expression node)
                     }
 
                     // Inspect the visited node
-                    this.Inspect();
+                    this.InspectAsync();
 
                     // Try to expand the visited node
                     // if it represents API data
                     if (this.context.ModelReference is DataSourceStubModelReference)
                     {
-                        node = this.Expand(visited);
+                        // Can not use await here as the method cannot be async
+                        node = this.ExpandAsync(visited).Result;
```

Comment:
These .Result negate the benefits you gain from making those calls async. You need to make the entire call chain async all the way up, or maybe just leave everything as is.